File: 2869. Minimum Operations to Collect Elements.py
class Solution(object):
    def minOperations(self, nums, k):
        num_set = set()
        step = 0
        for i in range(1, k+1):
            num_set.add(i)
        for i in range(len(nums)-1, -1, -1):
            if nums[i] in num_set:
                num_set.remove(nums[i])
            step += 1
            if len(num_set) == 0:
                return step
        

File: 2869. Minimum Operations to Collect Elements.ts
function minOperations(nums: number[], k: number): number {
let set = new Set(), step:number=0;
for (let i = 1; i <= k; i++) { set.add(i); }
for (let i=nums.length-1;i>=0;i--){
if (set.has(nums[i])) set.delete(nums[i]);
step++;
if (set.size==0) return step;
}
};
